### PR TITLE
Clean Up (see description for details)

### PR DESCRIPTION
**Pull request by kstenschke**
Removed duplicate variable declarations
Added missing semicolons
Removed unneccessary semicolons

### CODE REVIEW

**Comment by kadishmal** [comment]
@afshinm I think such contributions should be merged sooner. After all people have spent time cleaning up the mess. If you don't have time, I'm eager to help out.

**Comment by afshinm** [comment]
This is a great change. I will merge it for the next version. Thank you @kstenschke @kadishmal

File: intro.js [hunk 1]
```diff
@@ -457,12 +456,12 @@
...
-    var targetOffset = _getOffset(targetElement)
-    var tooltipHeight = _getOffset(tooltipLayer).height
-    var windowSize = _getWinSize()
+    var targetOffset = _getOffset(targetElement);
+    var tooltipHeight = _getOffset(tooltipLayer).height;
+    var windowSize = _getWinSize();
     switch (currentTooltipPosition) {
       case 'top':
-        tooltipLayer.style.left = '15px';
+        tooltipLayer.style.left = '15px';;
```
Comment (afshinm) [inline intro.js]: What happened here?

File: intro.js [hunk 2]
```diff
@@ -79,13 +79,12 @@
    */
   function _introForElement(targetElm) {
     var introItems = [],
-        self = this;
+        self = this,
+        i;
```
Comment (afshinm) [inline intro.js]: Please remove this `i` variable and lets define it in the for-loop.

File: intro.js [hunk 3]
```diff
@@ -717,9 +716,9 @@
       //remove `introjs-fixParent` class from the elements
       var fixParents = document.querySelectorAll('.introjs-fixParent');
       if (fixParents && fixParents.length > 0) {
-        for (var i = fixParents.length - 1; i >= 0; i--) {
```
Comment (afshinm) [inline intro.js]: The same thing here. Let's define the stepper inside for-loop.

File: intro.js [hunk 4]
```diff
@@ -1147,7 +1146,7 @@
   function _mergeOptions(obj1,obj2) {
     var obj3 = {};
     for (var attrname in obj1) { obj3[attrname] = obj1[attrname]; }
-    for (var attrname in obj2) { obj3[attrname] = obj2[attrname]; }
```
Comment (afshinm) [inline intro.js]: Same as previous comment.

**Comment by afshinm** [comment]
@kstenschke Can you please address mentioned issues?

File: intro.js (hunk 1 above)
Comment (kstenschke) [inline intro.js]: looks like a typo to me (i dont think this is a mistake during merging)

File: intro.js (hunk 4 above)
Comment (kstenschke) [inline intro.js]: you dont need to redeclare attrname over here.

**Comment by kstenschke** [comment]
@afshinm thank you for reviewing, i again addressed the mentioned issues (1 and 2: changed according to your suggestion, 3: commented)

File: intro.js (hunk 4 above)
Comment (afshinm) [inline intro.js]: I'm sorry but I can't get it. Why we shouldn't declare the variable using `var`? And why you think `attrname` is redeclared?

File: intro.js (hunk 1 above)
Comment (afshinm) [inline intro.js]: So please keep one of them.

File: intro.js [hunk 5]
```diff
@@ -124,7 +122,7 @@
       }
 
       //first add intro items with data-step
-      for (var i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
+      for (i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
```
Comment (afshinm) [inline intro.js]: Please use `var`

File: intro.js [hunk 6]
```diff
@@ -143,8 +141,8 @@
       //next add intro items without data-step
       //todo: we need a cleanup here, two loops are redundant
       var nextStep = 0;
-      for (var i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
-        var currentElement = allIntroSteps[i];
+      for (i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
```
Comment (afshinm) [inline intro.js]: Please use `var` for stepper `i`

File: intro.js [hunk 7]
```diff
@@ -143,8 +141,8 @@
...
+        currentElement = allIntroSteps[i];
```
Comment (afshinm) [inline intro.js]: Also, we should define `currentElement` within the for-loop, not globally. Use `var` here.

File: intro.js [hunk 8]
```diff
@@ -849,7 +847,7 @@
       referenceLayer.appendChild(tooltipLayer);
 
       //next button
-      var nextTooltipButton = document.createElement('a');
+      nextTooltipButton = document.createElement('a');
```
Comment (afshinm) [inline intro.js]: Need `var` here again.

**Comment by afshinm** [comment]
@kstenschke in your refactoring, you removed `var` in some lines. Can you please explain why? Can't understand the reason.

**Comment by kstenschke** [comment]
@afshinm i removed redundant var statements, that is: duplicate identical declarations of variables in a scope. Explanation: Once a variable is declared via var-statement in a scope, latter different value assignments should just define, not declare, that variable('s value alteration).

**Comment by afshinm** [comment]
Yes, right. But for some lines we shouldn't remove `var` since we didn't defined the variable before. I will mention them now.

Excuse me if I'm wasting your time, but it's really important to be accurate. I'm interested to merge your PR today so I can merge others.

**Comment by afshinm** [comment]
I merged this PR manually. Thank you @kstenschke

**Pull request closed without merging**